zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/message/DbProcessMessageSubscriptionState.java

```diff
@@ -54,18 +59,21 @@ public DbProcessMessageSubscriptionState(
       final TransactionContext transactionContext,
       final TransientPendingSubscriptionState transientProcessMessageSubscriptionState) {
     elementInstanceKey = new DbLong();
+    subscriptionKey = new DbLong();
     tenantIdKey = new DbString();
     messageName = new DbString();
     tenantAwareMessageName = new DbTenantAwareKey<>(tenantIdKey, messageName, PlacementType.PREFIX);
     elementKeyAndMessageName = new DbCompositeKey<>(elementInstanceKey, tenantAwareMessageName);
+    elementKeyMessageNameAndSubscriptionKey =
+        new DbCompositeKey<>(elementKeyAndMessageName, subscriptionKey);
     processMessageSubscription = new ProcessMessageSubscription();
     transientState = transientProcessMessageSubscriptionState;
 
     subscriptionColumnFamily =
         zeebeDb.createColumnFamily(
             ZbColumnFamilies.PROCESS_SUBSCRIPTION_BY_KEY,
             transactionContext,
-            elementKeyAndMessageName,
+            elementKeyMessageNameAndSubscriptionKey,
```

Comment:
💭 This change will require a migration task to append a `subscriptionKey` to each entry of the `PROCESS_SUBSCRIPTION_BY_KEY` column family. 

🆘 This could be a blocker because we don't have this key stored anywhere right now.

💡 Some ideas:
- generate new keys and somehow send these to the other partition. This will be hard because no migration task has had to communicate with other partitions before.
- generate new keys but don't send them to the other partition. When -1 is used as key, simply take the first entry based on the `elementInstanceKey`, and `tenantAwareMessageName`.
- don't generate new keys at all. When -1 is used as key, simply take the first entry based on the `elementInstanceKey`, and `tenantAwareMessageName`. This option might still require a migration task to append `-1` to the key. Or we should investigate whether we can read entries without the last 8 bytes (long value) as `-1`.

> [!NOTE]
> that key change migrations require iterating over all entries, and for each entry an insert and delete operation.

So it might not be a blocker, but comes either at an upgrade performance cost or at the cost of complex code dealing with this missing part of the key. I'm a bit worried about this part.